### ISSUE TITLE
AsciiDoc (or alternatives) instead of Markdown?

### ISSUE THREAD

**Issue by m-mohr** [ceos-org/ceos-ard#38]
I'm running into more and more issues with Markdown. The Markdown syntax is too simple or what we are trying to achieve here, especially with vasts amounts of tables.

We could consider using AsciiDoc or other similar alternatives instead.
It would also support includes, which allows re-use of commonly used elements.

Generally, I wonder whether the tabular approach is good or whether an approach based on headings and normal paragraphs would be better for such a specification.

_Originally posted by @m-mohr in https://github.com/ceos-org/ceos-ard/issues/28#issuecomment-2194242896_

**Comment by johntruckenbrodt** [comment]
How about restructured text? Standard for sphinx and much more versatile than markdown.

**Comment by m-mohr** [comment]
Could also be considered, indeed.

**Comment by avalentino** [comment]
Also for me reStructuredText is better suited for the kind of documents that you plan to manage

**Comment by m-mohr** [comment]
Could you elaborate a bit on the reasons to make us understand why it might be a better choice?

I don't have experience with any of them but they look very similar to me at first glance.

I find the file includes from AsciiDoc very appealing to avoid duplication, which seems to heavily appear on the files and as such would allow us to heavily decrease the maintanance work and move us closer to a biuilding block approach.

**Comment by avalentino** [comment]
Yes, as you saythe `include` directive is also available in reStructuredText.

To clarify, I was meaning that reStructuredText is better suited than markdown in this case.
I do not know AsciiDoc, sorry.

ReStructuredText, especially with [Sphinx](https://www.sphinx-doc.org) extension, is very powerful and supports very well complicit documents structure and cross-referencing.
Both the HTML output and the PDF are very good and can be highly customized (a large amount of HTML themes is available).
The tooling, Sphinx mostly, is very powerful and several extensions are available for different purposes.

**Comment by m-mohr** [comment]
Thanks.

(Just a note here: I believe we also still need Word/docx files for users to fill unless we generate separate documents for self-assessment and specification.)

**Comment by johntruckenbrodt** [comment]
Same as Antonio, very happy user of RST/Sphinx, don't know AsciiDoc.

**Comment by avalentino** [comment]
I have no big experience in generating docx document for rst/sphinx.
Searching with google the first hits are https://github.com/mherkazandjian/docxsphinx, https://github.com/amedama41/docxbuilder and a couple of indirect ways using rst2otd (docutils) and/or pandoc. I guess that there is also a possibility to make an indirect conversion via LaTeX but I never did it.

Just for my information, does asciidoc have better capabilities to convert to docx?

**Comment by m-mohr** [comment]
Not sure, that's something we need to evaluate, but Pandoc worked quite well to convert from Markdown to Word.

**Comment by johntruckenbrodt** [comment]
According to the [pandoc website](https://pandoc.org/), pandoc is capable of conversion from and to reStructuredText and docx respectively. AsciiDoc can only be converted to but not from.

**Comment by m-mohr** [comment]
from Siri Jodha Khalsa (via email):

> I just want to point out that some IEEE standards projects use [metanorma](https://www.metanorma.org/) to write their drafts, there being an IEEE-SA document model. Metanorma is based on AsciiDoc, which is similar to markdown ([comparison](https://docs.asciidoctor.org/asciidoc/latest/asciidoc-vs-markdown/)), all of which are amenable to version control in github. CEOS could create a custom stylesheet for the PFSs. See https://www.metanorma.org/develop/topics/simple-adoption/

**Comment by m-mohr** [comment]
Various possible solutions have been checked:

![Image](https://github.com/user-attachments/assets/df934b7c-440f-42bc-8702-0ebe423b4be7)

Metanorma wasn't even installable, issues were not answered by the metanorma team. I guess you only get a good solution there is you sign a support contract or so. Thus, this is not a solution right now.

The tech stack in general for generating Word and PDF is quite heavy in all cases.

For now we'll start with Pandoc and an extended Pandoc version of Markdown. This works pretty well right now and gives the lowest barrier of entry.

**Comment by avalentino** [comment]
Sorry @m-mohr, I see that you have not even considered solutions like Sphinx (https://www.sphinx-doc.org/en/master).

Sphinx is a widespread documentation build system that also implements an extended syntax on top of reStructuredText.

It give the possibility to build complex documents structures,  cross referencing between different files and automatically generating indices, TOCs, bibliography, etc., and, of course, supports images, math and tables (including labels and numbering).

It has the possibility to generate output in different formats (including HTML and PDF) and use user defined templates (including LaTeX ones).

Moreover the https://readthedocs.org/ site allows you to build and publish your documentation from the repository for each commit.

Please at least give it a try before deciding.

**Comment by m-mohr** [comment]
There's a requirement do build Word documents, which doesn't seem possible with your proposed tooling. Happy to be proven wrong @avalentino

**Comment by m-mohr** [comment]
Thanks. Via RTF, we seem to loose some formatting and docxsphinx and docxbuilder seem unmaintained.

The conversion via HTML had issues for other formats, but I can check again for sphinx output.

One drawback for RST comoared to Markdown is that it's known primarily in the Python community while Markdown is much more widespread across various communities.

**Comment by m-mohr** [comment]
I think the table above did a bad job explaining the choices, so I updated the table(s) to explain it better and reevaluated a couple of cells:

![Image](https://github.com/user-attachments/assets/84057548-de77-4c82-a117-83d1515d4b68)
